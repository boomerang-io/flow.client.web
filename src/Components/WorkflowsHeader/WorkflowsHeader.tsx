import React from "react";
import { Layer, Search, Breadcrumb, BreadcrumbItem } from "@carbon/react";
import {
  FeatureHeader as Header,
  FeatureHeaderTitle as HeaderTitle,
  FeatureHeaderSubtitle as HeaderSubtitle,
} from "@boomerang-io/carbon-addons-boomerang-react";
import CreateTemplateWorkflow from "Components/CreateTemplateWorkflow";
import { FlowTeam, WorkflowView, WorkflowViewType, Workflow } from "Types";
import { Link } from "react-router-dom";
import { appLink } from "Config/appConfig";
import styles from "./workflowsHeader.module.scss";

interface WorkflowsHeaderProps {
  pretitle?: React.ReactNode;
  title: string;
  subtitle: string;
  handleUpdateFilter: (args: { query: string }) => void;
  searchQuery: string | string[] | null;
  team?: FlowTeam | null;
  workflowList: Array<Workflow>;
  viewType: WorkflowViewType;
}

const WorkflowsHeader: React.FC<WorkflowsHeaderProps> = ({
  pretitle,
  title,
  subtitle,
  handleUpdateFilter,
  searchQuery,
  team,
  workflowList,
  viewType,
}) => {
  const workflowsCount = workflowList.length;
  const workflowsCountStr = workflowsCount > 0 ? `(${workflowsCount})` : "";

  const handleOnSearchInputChange = (e: React.MouseEvent<HTMLInputElement, MouseEvent>) => {
    handleUpdateFilter({ query: e.currentTarget?.value ?? "" });
  };

  const NavigationComponent = () => {
    return (
      <Breadcrumb noTrailingSlash>
        <BreadcrumbItem>
          <Link to={appLink.home()}>Home</Link>
        </BreadcrumbItem>
        <BreadcrumbItem isCurrentPage>
          <p>{team ? team.displayName : "---"}</p>
        </BreadcrumbItem>
      </Breadcrumb>
    );
  };

  return (
    <Header
      className={styles.container}
      includeBorder={false}
      nav={viewType === WorkflowView.Workflow ? <NavigationComponent /> : null}
      header={
        <>
          {Boolean(pretitle) ? <HeaderSubtitle>{pretitle}</HeaderSubtitle> : null}
          <HeaderTitle>{`${title} ${workflowsCountStr}`}</HeaderTitle>
          {Boolean(subtitle) ? <HeaderSubtitle className={styles.headerMessage}>{subtitle}</HeaderSubtitle> : null}
        </>
      }
      actions={
        <ActionsBar>
          {viewType === WorkflowView.Workflow ? <CreateTemplateWorkflow team={team!} workflowList={[]} /> : null}
          <Layer className={styles.search}>
            <Search
              disabled={!workflowsCount || workflowsCount === 0}
              data-testid="workflows-team-search"
              id="search-team-workflows"
              labelText={`Search for a ${viewType}`}
              onChange={handleOnSearchInputChange}
              placeholder={`Search for a ${viewType}`}
              value={searchQuery}
            />
          </Layer>
        </ActionsBar>
      }
    />
  );
};

export default WorkflowsHeader;

interface ActionsBarProps {
  children: React.ReactNode;
}

const ActionsBar: React.FC<ActionsBarProps> = (props: ActionsBarProps) => {
  return <div className={styles.filterContainer}>{props.children}</div>;
};
