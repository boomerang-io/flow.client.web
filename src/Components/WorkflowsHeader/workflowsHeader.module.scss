.container {
  h1 {
    margin: 0;
  }

  h2 {
    font-size: 1.25rem;
    font-weight: 300;
  }

  :global .bx--search-input {
    background: var(--cds-ui-background);
    font-size: 0.875rem;
    height: 3rem;
  }
  :global .b-search-filter {
    align-items: flex-end;
    align-self: flex-end;
    width: 36rem;
  }

  :global .b-search-filter__filter {
    font-size: 0.875rem;
    .bx--list-box {
      background-color: var(--cds-ui-background) !important;
      max-height: none !important;
    }

    .bx--list-box__menu {
      background-color: var(--cds-ui-background);
    }
    .bx--list-box__menu-item {
      background: var(--cds-ui-background);
    }
    .bx--list-box__menu-item--highlighted {
      background: var(--cds-ui-background);
    }
  }
}

.headerMessage {
  margin-top: 0.5rem;
}

.filterContainer {
  align-items: flex-end;
  display: flex;
  margin-bottom: 1.5rem;
}

.search {
  flex: 1 1 auto;
  margin-right: 2rem;

  & .bx--search--lg.bx--search-input {
    height: 2.75rem !important;
  }
}

.filter {
  width: 15rem;
  :global .bx--list-box {
    background: var(--cds-ui-background);
  }
}
