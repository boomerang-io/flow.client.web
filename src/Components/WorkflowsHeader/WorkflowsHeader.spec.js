import React from "react";
import WorkflowsHeader from "./index";

const mockfn = jest.fn();

const props = {
  workflowsLength: 1,
  teamsQuery: [],
  handleSearchFilter: mockfn,
  isLoading: false,
  options: [
    {
      name: "test team",
      id: "testid",
    },
  ],
};

describe("WorkflowsHeader --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowsHeader", () => {
    const { baseElement } = rtlRouterRender(<WorkflowsHeader {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
