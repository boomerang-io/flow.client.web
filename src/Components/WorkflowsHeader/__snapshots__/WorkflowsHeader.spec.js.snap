// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowsHeader --- Snapshot Capturing Snapshot of WorkflowsHeader 1`] = `
<body
  id="app"
>
  <div>
    <header
      class="bx--bmrg-feature-header container"
    >
      <section
        class="bx--bmrg-feature-header__content"
      >
        <div
          class="bx--bmrg-feature-header__header"
        >
          <p
            class="bx--bmrg-feature-header-title"
          >
            These are your
          </p>
          <h1
            class="bx--bmrg-feature-header-title"
          >
            Workflows (undefined)
          </h1>
        </div>
        <div
          class="bx--bmrg-feature-header__footer"
        >
          <nav
            class="bx--bmrg-feature-nav-tabs"
          >
            <a
              class="bx--bmrg-feature-nav-tab"
              href="/workflows/mine"
            >
              My Workflows
            </a>
            <a
              class="bx--bmrg-feature-nav-tab"
              href="/workflows/teams"
            >
              Team Workflows
            </a>
          </nav>
        </div>
      </section>
      <div
        class="filterContainer"
      >
        <div
          class="search"
        >
          <div
            aria-labelledby="search-team-workflows-search"
            class="bx--search bx--search--xl"
            role="search"
          >
            <svg
              aria-hidden="true"
              class="bx--search-magnifier"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5	C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
              />
            </svg>
            <label
              class="bx--label"
              for="search-team-workflows"
              id="search-team-workflows-search"
            >
              Search for a workflow
            </label>
            <input
              autocomplete="off"
              class="bx--search-input"
              data-testid="workflows-team-search"
              id="search-team-workflows"
              placeholder="Search for a workflow"
              role="searchbox"
              type="text"
              value=""
            />
            <button
              aria-label="Clear search input"
              class="bx--search-close bx--search-close--hidden"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>
</body>
`;
